src/sampling.jl

```diff
@@ -36,12 +71,9 @@ and set `x[j] = a[i]`, with `n=length(a)` and `k=length(x)`.
 This algorithm consumes `k` random numbers.
 """
 function direct_sample!(rng::AbstractRNG, a::AbstractArray, x::AbstractArray)
-    1 == firstindex(a) == firstindex(x) ||
-        throw(ArgumentError("non 1-based arrays are not supported"))
-    Base.mightalias(a, x) &&
-        throw(ArgumentError("output array x must not share memory with input array a"))
-    s = Sampler(rng, 1:length(a))
-    for i = 1:length(x)
+    n, k = _validate_sample_inputs(a, x, true)
+    s = Sampler(rng, 1:n)
+    for i = 1:k
```

Comment:
Is it really an improvement to return `n, k` from `_validate_sample_inputs`? This makes the code harder to understand and the performance gain is probably negligible compared to the cost of `mightalias`.